ext/php_cassandra.c

```diff
@@ -173,12 +174,67 @@ ZEND_GET_MODULE(cassandra)
 #endif
 
 static void
-php_cassandra_log(const CassLogMessage* message, void *data)
+php_cassandra_log(const CassLogMessage* message, void* data)
 {
-  fprintf(stderr, "php-driver | [%s] %s (%s:%d)\n",
-    cass_log_level_string(message->severity), message->message,
-    message->file, message->line
-  );
+  char log[MAXPATHLEN + 1];
+  uint log_length = 0;
+
+  (void)data;
+
+  uv_rwlock_rdlock(&CASSANDRA_G(log_lock));
+  if (CASSANDRA_G(log)) {
+    log_length = MIN(CASSANDRA_G(log_length), MAXPATHLEN);
+    memcpy(log, CASSANDRA_G(log), log_length);
+  }
+  uv_rwlock_rdunlock(&CASSANDRA_G(log_lock));
+
+  log[log_length] = '\0';
+
+  if (log_length > 0) {
+    int fd = -1;
+
+    if (!strcmp(log, "syslog")) {
+      php_syslog(LOG_NOTICE, "php-driver | [%s] %s (%s:%d)",
+                 cass_log_level_string(message->severity), message->message,
+                 message->file, message->line);
+      return;
+    }
+
+    fd = VCWD_OPEN_MODE(log, O_CREAT | O_APPEND | O_WRONLY, 0644);
+
+    if (fd != 1) {
+      time_t log_time;
+      struct tm log_tm;
+      char log_time_str[32];
+
+      time(&log_time);
+      php_localtime_r(&log_time, &log_tm);
+      strftime(log_time_str, sizeof(log_time_str), "%d-%m-%Y %H:%M:%S %Z", &log_tm);
+
+      size_t needed = snprintf(NULL, 0, "%s [%s] %s (%s:%d)%s",
+                               log_time_str,
+                               cass_log_level_string(message->severity), message->message,
+                               message->file, message->line,
+                               PHP_EOL);
+
+      char* tmp = malloc(needed + 1);
+      sprintf(tmp, "%s [%s] %s (%s:%d)%s",
+              log_time_str,
+              cass_log_level_string(message->severity), message->message,
+              message->file, message->line,
+              PHP_EOL);
+
+      write(fd, tmp, needed);
+      free(tmp);
+      close(fd);
+      return;
+    }
+  }
+
+  fprintf(stderr, "php-driver | [%s] %s (%s:%d)%s",
```

Comment:
I guess thread-safety would be a concern. We could default to sapi if threading not enabled and write to stderr otherwise, what do you think?